### ISSUE TITLE
Error generating for the second time

### ISSUE THREAD

**Issue by rogeriodeoliveira** [lllyasviel/Fooocus#586]
The first time the fooocus is opened, the images are generated perfectly, the second attempt to generate, the following error occurs:

```
D:\Programas\Fooocus_win64_2-1-14>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Fast-forward merge
Update succeeded.
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.1.19
Inference Engine exists and URL is correct.
Inference Engine checkout finished for 1c5d6663faf1a33e00ec67240167b174a9cac655.
Total VRAM 4096 MB, total RAM 15741 MB
Trying to enable lowvram mode because your GPU seems to have 4GB or less. If you don't want this use: --normalvram
xformers version: 0.0.20
Set vram state to: LOW_VRAM
Device: cuda:0 NVIDIA GeForce GTX 1650 : native
VAE dtype: torch.float32
Using xformers cross attention
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
model_type EPS
adm 2560
Refiner model loaded: D:\Programas\Fooocus_win64_2-1-14\Fooocus\models\checkpoints\sd_xl_refiner_1.0_0.9vae.safetensors
model_type EPS
adm 2816
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
missing {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids'}
Base model loaded: D:\Programas\Fooocus_win64_2-1-14\Fooocus\models\checkpoints\sd_xl_base_1.0_0.9vae.safetensors
LoRAs loaded: [('sd_xl_offset_example-lora_1.0.safetensors', 0.5), ('None', 0.5), ('None', 0.5), ('None', 0.5), ('None', 0.5)]
Fooocus Expansion engine loaded for cuda:0, use_fp16 = False.
loading new
loading in lowvram mode 1758.076922416687
loading in lowvram mode 418.90662479400635
App started successful. Use the app with http://127.0.0.1:7860/ or 127.0.0.1:7860
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 7.0
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 20
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
model_type EPS
adm 2816
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
missing {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_g.logit_scale', 'cond_stage_model.clip_l.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale'}
Base model loaded: D:\Programas\Fooocus_win64_2-1-14\Fooocus\models\checkpoints\juggernautXL_version5.safetensors
LoRAs loaded: [('None', 0.5), ('None', 0.5), ('None', 0.5), ('None', 0.5), ('None', 0.5)]
loading new
loading in lowvram mode 1438.3676376342773
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] New suffix: extremely detailed eyes, fantastic details full face, mouth, trending on artstation, pixiv, cgsociety, hyperdetailed Unreal Engine 4k 8k ultra HD, Stanley Artgerm Lau, WLOP, Rossdraws, James Jean Marc Simonetti Ruan Jia and Mandy Jurgens and Artgerm and William-Adolphe Bouguerea Sakimichan
[Fooocus] Preparing Fooocus text #2 ...
[Prompt Expansion] New suffix: intricate, elegant, highly detailed, digital painting, artstation, concept art, smooth, sharp focus, illustration, art by artgerm and greg rutkowski and alphonse mucha
[Fooocus] Encoding positive #1 ...
[Fooocus] Encoding positive #2 ...
[Fooocus] Encoding negative #1 ...
[Fooocus] Encoding negative #2 ...
Preparation time: 35.29 seconds
loading new
loading in lowvram mode 1737.9807691574097
Moving model to GPU: 4.91 seconds
  0%|                                                                                                        | 0/30 [00:00<?, ?it/s]D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torchsde\_brownian\brownian_interval.py:594: UserWarning: Should have tb<=t1 but got tb=14.614643096923828 and t1=14.614643.
  warnings.warn(f"Should have {tb_name}<=t1 but got {tb_name}={tb} and t1={self._end}.")
 93%|████████████████████████████████████████████████████████████████████████████████████████▋      | 28/30 [05:45<00:22, 11.17s/it]D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torchsde\_brownian\brownian_interval.py:585: UserWarning: Should have ta>=t0 but got ta=0.02916753850877285 and t0=0.029168.
  warnings.warn(f"Should have ta>=t0 but got ta={ta} and t0={self._start}.")
100%|███████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [06:08<00:00, 12.28s/it]
Image generated with private log at: D:\Programas\Fooocus_win64_2-1-14\Fooocus\outputs\2023-10-08\log.html
Generating and saving time: 376.53 seconds
loading new
loading in lowvram mode 1721.4047470092773
100%|███████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [05:47<00:00, 11.59s/it]
Image generated with private log at: D:\Programas\Fooocus_win64_2-1-14\Fooocus\outputs\2023-10-08\log.html
Generating and saving time: 358.10 seconds
loading new
loading in lowvram mode 1712.8605766296387
Total time: 777.66 seconds
loading in lowvram mode 412.954852104187
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 7.0
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 20
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
Traceback (most recent call last):
  File "D:\Programas\Fooocus_win64_2-1-14\Fooocus\modules\async_worker.py", line 552, in worker
    handler(task)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\Programas\Fooocus_win64_2-1-14\Fooocus\modules\async_worker.py", line 263, in handler
    expansion = pipeline.final_expansion(prompt, t['task_seed'])
  File "D:\Programas\Fooocus_win64_2-1-14\Fooocus\modules\expansion.py", line 64, in __call__
    features = self.model.generate(**tokenized_kwargs,
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\transformers\generation\utils.py", line 1572, in generate
    return self.sample(
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\transformers\generation\utils.py", line 2619, in sample
    outputs = self(
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\accelerate\hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 1080, in forward
    transformer_outputs = self.transformer(
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 903, in forward
    outputs = block(
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 391, in forward
    attn_outputs = self.attn(
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 332, in forward
    attn_output, attn_weights = self._attn(query, key, value, attention_mask, head_mask)
  File "D:\Programas\Fooocus_win64_2-1-14\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 203, in _attn
    attn_weights = torch.where(causal_mask, attn_weights.to(attn_weights.dtype), mask_value)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
Total time: 1.02 seconds
```

It is always necessary to finish the fooocus and open it again to work again only for the first time.

Hardware:
AMD Ryzen 7 4800H with Radeon Graphics, 2900 Mhz
Nvidia GeForce GTX 1650 4Gb
RAM: 16,0 GB
SSD free space 530GB
Windows 11 Pro

**Comment by lllyasviel** [comment]
give full log including system info from beginning

**Comment by lllyasviel** [comment]
see if fixed in 2.1.21

**Comment by rogeriodeoliveira** [comment]
> see if fixed in 2.1.21

Yes, issue fixed in 2.1.21
